sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/ParallelSecretLoader.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Security.KeyVault.Secrets;
+
+namespace Azure.Extensions.AspNetCore.Configuration.Secrets
+{
+    internal class ParallelSecretLoader: IDisposable
+    {
+        private const int ParallelismLevel = 32;
```

Comment:
Is there any practical chance more than one instance is created? For example, could middleware in the app pipeline fetch it's own? Is that enough of a concern to basically design this class to be static and reentrant? You could, for example, keep queueing secrets to fetch and have a pool of worker limited worker threads dequeueing from it (a tactic I've used in the past on a couple different projects).

Reply:
More than one instance of this class? I expect it's possible but not too common as you can add multiple KeyVaults to a configuration builder.

I'm a bit reluctant to create a global chokepoint though as those have their own set of scaling issues.